client/client.go

```diff
@@ -56,6 +56,36 @@ import (
 	"github.com/pkg/errors"
 )
 
+// DummyHost is a hostname used for local communication.
+//
+// It acts as a valid formatted hostname for local connections (such as "unix://"
+// or "npipe://") which do not require a hostname. It should never be resolved,
+// but uses the special-purpose ".localhost" TLD (as defined in [RFC 2606, Section 2]
```

Comment:
Would it make sense (just in case) to use `invalid.` TLD that explicitly prohibits resolution while `localhost` typically resolves to loopback IP?

From https://www.rfc-editor.org/rfc/rfc2606.html#section-2

    ".invalid" is intended for use in online construction of domain
    names that are sure to be invalid and which it is obvious at a
    glance are invalid.
    
    The ".localhost" TLD has traditionally been statically defined in
    host DNS implementations as having an A record pointing to the
    loop back IP address and is reserved for such use.  Any other use
    would conflict with widely deployed code which assumes this use. 

From https://www.rfc-editor.org/rfc/rfc6761#section-6.4

    The domain "invalid." and any names falling within ".invalid." are
    special in the ways listed below.  In the text below, the term
    "invalid" is used in quotes to signify such names, as opposed to
    names that may be invalid for other reasons (e.g., being too long).

Reply:
E.g. `this.placeholder.host.is.intentionally.invalid`

Reply:
I was playing with the idea to use `.invalid`, but considered that it may be potentially confusing to use an "invalid" host as a "valid" hostname. I'm not 100% attached to that though, and now that we have a `const` for this, it would also be easy to change it later if we feel like doing so.

Perhaps one of Cory's ideas (`.` or `-`) could then also be considered.